### PR TITLE
Modtool reports actions

### PR DESCRIPTION
**Pull request by scarow**
https://reddit.atlassian.net/browse/CE-369
https://reddit.atlassian.net/browse/CE-382
https://reddit.atlassian.net/browse/CE-534 (add reports modal -- just combined the PRs)

https://modtool-reports-actions.mobile.staging.snooguts.net

Note that when a post gets approved, the reports are cleared on refresh.
Also note that approved posts can not be reported (you can technically report them from the interface, but a report flag will not show up).

### CODE REVIEW

**Comment by scarow** [comment]
👓 @birakattack @madbook @powerlanguage 

Josh can you read the behavior above and tell me that's expected/needs to change?

File: src/app/components/Post/PostHeader/index.jsx [hunk 1]
```diff
@@ -169,6 +169,12 @@ function renderAuthorAndTimeStamp(post, single, hideWhen, className) {
   );
 }
 
+function sumReportsCount(userReports, modReports) {
+  let reportCounts = userReports.map(r => r[1]);
+  reportCounts.push(modReports.length);
+  return reportCounts.reduce((a, b) => a + b, 0);
+}
+
```
Comment (scarow) [inline src/app/components/Post/PostHeader/index.jsx]: This has been moved into a helper in a subsequent PR

**Comment by powerlanguage** [comment]
@scarow the behavior mirrors desktop.  Agree it feels weird, but I think it is fine for this version.

**Comment by scarow** [comment]
Actually -- y'all can ignore this. New PR that includes both comments and posts: https://github.com/reddit/reddit-mobile/pull/1004

**Pull request closed without merging**